src/Modules/OrchardCore.Commerce/Services/StripePaymentService.cs

```diff
@@ -170,6 +176,11 @@ public async Task UpdateOrderToOrderedAsync(PaymentIntent paymentIntent)
             await workflowManager.TriggerEventAsync(nameof(OrderCreatedEvent), order, "Order-" + order.ContentItemId);
         }
 
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+
+        // Decrease inventories of purchased items.
+        await _productInventoryService.UpdateInventoriesAsync(currentShoppingCart.Items);
```

Comment:
Moved this call here to address bug where inventories were decreased even after unsuccessful payment.